test/control-plane/node/ciliumnodes/ciliumnodes_validation.go

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package node
+
+import (
+	"errors"
+	"fmt"
+	"path"
+	"reflect"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/cilium/cilium/pkg/datapath/fake"
+	v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	controlplane "github.com/cilium/cilium/test/control-plane"
+)
+
+type goldenCiliumNodesValidator struct {
+	step           int
+	expectedLabels map[string]string
+}
+
+func NewGoldenCiliumNodesValidator(stateFile string, update bool) controlplane.Validator {
+	var step int
+	fmt.Sscanf(path.Base(stateFile), "state%d.yaml", &step)
+
+	switch step {
+	case 1:
```

Comment:
nit: I personally would prefer a descriptive string here as the "step" identifier, but that's a personal preference and not a blocking comment.

Reply:
I'm following up soon with a refactoring of the test framework that will do away with this.

Reply:
E.g. the tests will look more like this: https://github.com/joamaki/cilium/commit/043f2cf66ab84cc40cbbb1a50b6a143c4424708b#diff-d3ea0b01ee85e6abd9fcd62e5b933d2c977f8a32fe58213e075486d922571dca